### PR TITLE
fix goal line on scatter plot

### PR DESCRIPTION
**Pull request by EmmadUsmani**
Closes https://github.com/metabase/metabase/issues/36584

### Description

Adding a goal line to the scatter plot caused an extra line series to appear. This PR fixes that by making that series invisible.

The fix is in `goal-line.ts`, the rest the PR is just adding a storybook example.

### How to verify

1. Create any scatter plot
2. Enable goal line in viz settings
3. Add to dashboard and send a subscription
4. Confirm nothing extra besides the scatter points and goal line appear on the chart

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Comment by EmmadUsmani** [comment]
Current dependencies on/for this PR:
* master
  * **PR #36489** <a href="https://app.graphite.dev/github/pr/metabase/metabase/36489?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #36486** <a href="https://app.graphite.dev/github/pr/metabase/metabase/36486?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #36135** <a href="https://app.graphite.dev/github/pr/metabase/metabase/36135?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #36563** <a href="https://app.graphite.dev/github/pr/metabase/metabase/36563?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #36317** <a href="https://app.graphite.dev/github/pr/metabase/metabase/36317?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #36502** <a href="https://app.graphite.dev/github/pr/metabase/metabase/36502?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #36510** <a href="https://app.graphite.dev/github/pr/metabase/metabase/36510?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #36560** <a href="https://app.graphite.dev/github/pr/metabase/metabase/36560?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #36548** <a href="https://app.graphite.dev/github/pr/metabase/metabase/36548?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #36564** <a href="https://app.graphite.dev/github/pr/metabase/metabase/36564?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #36013** <a href="https://app.graphite.dev/github/pr/metabase/metabase/36013?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35997** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35997?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35923** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35923?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35922** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35922?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35783** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35783?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #35996** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35996?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35573** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35573?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35395** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35395?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35345** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35345?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35321** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35321?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35307** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35307?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35304** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35304?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35303** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35303?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35301** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35301?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35300** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35300?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35297** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35297?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35118** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35118?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #34820** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34820?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #34923** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34923?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #34717** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34717?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #34772** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34772?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #34809** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34809?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #34816** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34816?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #34859** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34859?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #34869** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34869?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **PR #34872** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34872?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                    * **PR #35068** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35068?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #34703** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34703?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #34763** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34763?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/metabase/metabase/36564?utm_source=stack-comment).

File: frontend/src/metabase/visualizations/echarts/cartesian/option/goal-line.ts [hunk 1]
```diff
@@ -16,7 +16,14 @@ export function getGoalLineSeriesOption(
   const [_leftAxisKeys, rightAxisKeys] = chartModel.yAxisSplit;
 
   return {
-    type: "line",
+    type: "line", // type is irreelevant since we hide the series data with opacity 0
```
Comment (alxnddr) [inline frontend/src/metabase/visualizations/echarts/cartesian/option/goal-line.ts:19]: Did not test it here, but maybe `data: []` would work here — I use it for the timeline events series

Comment (EmmadUsmani) [reply frontend/src/metabase/visualizations/echarts/cartesian/option/goal-line.ts:19]: That makes sense, I'll use that

Comment (EmmadUsmani) [reply frontend/src/metabase/visualizations/echarts/cartesian/option/goal-line.ts]: Done

**Review by alxnddr — APPROVED** [review]

**Pull request merged**